firefox-ios/Providers/RustSyncManager.swift

```diff
@@ -166,6 +165,19 @@ public class RustSyncManager: NSObject, SyncManager {
         backgrounded = true
     }
 
+    private func setPreferenceForSignIn() {
+        let signedInFxaAccountValue = profile?.prefs.boolForKey(PrefsKeys.Sync.signedInFxaAccount)
+
+        guard signedInFxaAccountValue == nil else { return }
```

Comment:
we only want to set it if its nil, there is a case that the user has a syncable account, but returns false so that's why I had to check if its nil first.

Reply:
Should we add a comment here too? 🤔

Reply:
good point! if I feel I need to add a comment in the PR, I'll add it in the code as well in the future. thanks :)